restler/engine/core/request_utilities.py

```diff
@@ -50,56 +54,71 @@ def str_to_hex_def(val_str):
     """
     return hashlib.sha1(val_str.encode(UTF8)).hexdigest()
 
-def execute_token_refresh_cmd(cmd):
-    """ Forks a subprocess to execute @param cmd to refresh token.
-
-    @param cmd: The user-provided command to refresh the token.
-    @type  cmd: Str
-
-    @return: The result of the command
-    @rtype : Str
 
-    """
+def execute_token_refresh(token_dict):
     global latest_token_value, latest_shadow_token_value
-    _RAW_LOGGING(f"Will refresh token: {cmd}")
-
-    MAX_RETRIES = 5
-    RETRY_SLEEP_TIME_SEC = 2
     ERROR_VAL_STR = 'ERROR\r\n'
-    retry_count = 0
-    while retry_count < MAX_RETRIES:
-        try:
-            if sys.platform.startswith('win'):
-                cmd_result = subprocess.getoutput(str(cmd).split(' '))
-            else:
-                cmd_result = subprocess.getoutput([cmd])
+    result = None
+    token_auth_method = token_dict["token_auth_method"]
 
-            _RAW_LOGGING(f"New value: {cmd_result}")
-            _, latest_token_value, latest_shadow_token_value = parse_authentication_tokens(cmd_result)
-            _RAW_LOGGING(f"Successfully obtained the latest token")
+    retry_handler = RetryHandler()
+
+    while retry_handler.can_retry():
+        try:
+            if token_auth_method == "location":
```

Comment:
NIT: constant strings vs string

Reply:
Agreed, could you please create an enum for these?

Reply:
sure, for the token auth methods? Do you have a suggestion on where to define them

Reply:
Can you put them in restler_settings.py?

Reply:
added